pkg/storage/streams/store.go

```diff
@@ -100,6 +101,21 @@ func (s *streamStore) Put(ctx context.Context, path paths.Path, data io.Reader,
 	metadata []byte, expiration time.Time) (m Meta, err error) {
 	defer mon.Task()(&ctx)(&err)
 
+	// if previously same file uploaded, then clean up
+	lastSegmentMeta, err := s.segments.Meta(ctx, path.Prepend("l"))
+
+	// previously file uploaded
+	if err == nil {
```

Comment:
I don't know if this check is enough, you may want to also check for mutation on the identity of the lastSegmentMeta's value?

Reply:
we together went over the condition.